pyresample/utils/cf.py

```diff
@@ -64,15 +64,15 @@
 def _convert_XY_CF_to_Proj(crs, axis_info):
     """Convert XY values from CF to PROJ convention. With CF =< 1.9 only affects geostrationary projection."""
     crs_dict = crs.to_dict()
-    if crs_dict['proj'] == 'geos':
+    if (crs_dict['proj'] == 'geos') and (axis_info.get('unit') or 'radians' == 'radians'):
```

Comment:
```suggestion
    axis_units = axis_info.get('unit') or 'radians'  # unit could be None
    if crs_dict['proj'] == 'geos' and axis_units == 'radians':
```

Going to try to make deepcode happy.